File: daily/199-Binary-tree-right-side-view.py
class Solution:
    def rightSideView(self, root: Optional[TreeNode]) -> List[int]:
        if not root:
            return []
        result = []
        queue = deque([root])
        while queue:
            level_length = len(queue)
            for i in range(level_length):
                node = queue.popleft()
                if i == level_length - 1:
                    result.append(node.val)
                if node.left:
                    queue.append(node.left)
                if node.right:
                    queue.append(node.right)
        return result

File: daily/Binary Tree/199-Binary-tree-right-side-view.go
package daily

func rightSideView(root *TreeNode) []int {
	result := []int{}
	if root == nil {
		return []int{}
	}
	queue := []*TreeNode{root}
	for len(queue) > 0 {
		level_length := len(queue)
		for i := range level_length {
			node := queue[0]
			queue = queue[1:]
			if i == level_length-1 {
				result = append(result, node.Val)
			}
			if node.Left != nil {
				queue = append(queue, node.Left)
			}
			if node.Right != nil {
				queue = append(queue, node.Right)
			}
		}
	}
	return result
}
